service-base/src/test/java/org/eclipse/hono/service/metric/MicrometerBasedMetricsTest.java

```diff
@@ -268,52 +273,155 @@ public void testTimeoutEvent(final MeterRegistry registry) {
+        Thread.sleep(2L);
+        timerHandler.get().handle(null);
+
+        // THEN the metrics have been removed
+        assertNull(registry.find(MicrometerBasedMetrics.METER_MESSAGES_PAYLOAD).tags(tenantTags).meter());
+        assertNull(registry.find(MicrometerBasedMetrics.METER_MESSAGES_RECEIVED).tags(tenantTags).meter());
+        assertNull(registry.find(MicrometerBasedMetrics.METER_COMMANDS_PAYLOAD).tags(tenantTags).meter());
+        assertNull(registry.find(MicrometerBasedMetrics.METER_COMMANDS_RECEIVED).tags(tenantTags).meter());
+        assertNull(registry.find(MicrometerBasedMetrics.METER_CONNECTIONS_AUTHENTICATED).tags(tenantTags).meter());
 
-        // and later a second message
+    }
+
+    /**
+     * Verifies that sending messages updates the stored timestamp for the tenant.
+     * 
+     * @param registry The registry that the tests should be run against.
+     * @throws InterruptedException if thread sleep is interrupted.
+     */
+    @ParameterizedTest
+    @MethodSource("registries")
+    public void testLastSeenIsUpdatedOnMessageSend(final MeterRegistry registry) throws InterruptedException {
+
+        // GIVEN a metrics instance with tenantIdleTimeout configured
+        final MicrometerBasedMetrics metrics = new MicrometerBasedMetrics(registry, mock(Vertx.class));
+        metrics.setProtocolAdapterProperties(configWithTenantIdleTimeout(10L));
+
+        // WHEN connecting
+        metrics.incrementConnections(tenant);
+        final long before = metrics.getLastSeenTimestampPerTenant().get(tenant);
+
+        // and later sending a message
         Thread.sleep(1L);
         reportTelemetry(metrics);
 
         // THEN the timestamp for the tenant has been updated
-        assertEquals(1, metrics.getLastSendTimestampsPerTenant().size());
-        assertNotEquals(timestampOfFirstMessage, metrics.getLastSendTimestampsPerTenant().get(tenant));
+        assertEquals(1, metrics.getLastSeenTimestampPerTenant().size());
+        assertNotEquals(before, metrics.getLastSeenTimestampPerTenant().get(tenant));
+    }
+
+    /**
+     * Verifies that disconnecting updates the stored timestamp for the tenant.
+     * 
+     * @param registry The registry that the tests should be run against.
+     * @throws InterruptedException if thread sleep is interrupted.
+     */
+    @ParameterizedTest
+    @MethodSource("registries")
+    public void testLastSeenIsUpdatedOnDisconnect(final MeterRegistry registry) throws InterruptedException {
+
+        // GIVEN a metrics instance with tenantIdleTimeout configured
+        final MicrometerBasedMetrics metrics = new MicrometerBasedMetrics(registry, mock(Vertx.class));
+        metrics.setProtocolAdapterProperties(configWithTenantIdleTimeout(10L));
+
+        // WHEN connecting
+        metrics.incrementConnections(tenant);
+        final long before = metrics.getLastSeenTimestampPerTenant().get(tenant);
+
+        // and later disconnect
+        Thread.sleep(1L);
```

Comment:
given that you have full access to the map from the test case, I'd rather initialize the last seen timestamp to e.g. 0 instead of sleeping for a second here ...

Reply:
It is a millisecond :) But yes, sleeping is not optimal.